### PR TITLE
MM-22289 Fix for toast appearing when scrolled up

### PR DESCRIPTION
**Pull request by sudheerDev**
#### Summary
```
typeof showUnreadToast === 'undefined' && !props.atBottom
```
The above lines were used to set the initial state of toast but if toast state is never set and scrolled up the above lines execute causing the toast to show up.

  * The core issue of the problem is that toast state should be decided after the initial Scroll Correction is done. There was no way to determine before this causing an issue on when to set the state for toast.

  *~~Adding a state in virt list to know when init scroll is completed. So using this we set the initial state of toast so that when scrolling up the~~
 This above method is having a problem with some race i.e initScrollCompleted is not getting set at times so, causing it to get set on the next load of posts showing the toast when scrolled up in specific cases. Instead reverting to a logic of depending on the value of atBottom to verify by initially setting it to null

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-22289

### CODE REVIEW

**Comment by jgilliam17** [comment]
@sudheerDev 
I am still seeing the toast when opening an unread channel at the bottom and scrolling up

Feb-06-2020 10-30-59

**Comment by sudheerDev** [comment]
@jgilliam17 Can you check if you can replicate this? I am unable to replicate this

**Comment by jgilliam17** [comment]
> @jgilliam17 Can you check if you can replicate this? I am unable to replicate this

Same here, I'm no longer able to reproduce.

**Comment by jgilliam17** [comment]
@sudheerDev 
Per our discussion I re-tested try by creating different channels with 15 or more posts 
I was able to recreate with 30 posts.
1. start with brand 'new' public channel
2. user 1 posts `/test posts fuzz 30` to new channel 
3. sysadmin views all posts in the new channel and switches to different channel 
4. user 1 post  1 test message in `new` channel 
5. sysadmin clicks on the 'new` channel, sees the latest post on the bottom of the channel and scrolls all the way up to the top.

Observed: toast appears

**Comment by sudheerDev** [comment]
@Willyfrog Had to change the logic a bit. Mind giving this another look?

File: components/post_view/post_list_virtualized/post_list_virtualized.jsx [hunk 1]
```diff
@@ -125,7 +125,9 @@ class PostList extends React.PureComponent {
         this.state = {
             isScrolling: false,
             isMobile,
-            atBottom: true,
+
+            /* Intentionally setting undefined so that toast can determine when the first time this state is defined */
```
[the remark this answers is not in the document]
Comment (sudheerDev) [reply components/post_view/post_list_virtualized/post_list_virtualized.jsx]: Nice call, Changed to `null`

**Comment by sudheerDev** [comment]
> looking good, but needs to fix the tests!

I thought i pushed the commit fixing it yesterday. Never mind i did that earlier today though.

**Pull request merged**